### PR TITLE
ESC-250 passthrough for delete

### PR DESCRIPTION
**Pull request by sam-miele**

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -9,4 +9,5 @@ POST       /undertaking/member/:undertakingRef              uk.gov.hmrc.eusubsid
 POST       /undertaking/member/remove/:undertakingRef       uk.gov.hmrc.eusubsidycompliance.controllers.UndertakingController.deleteMember(undertakingRef)
 
 POST       /subsidy/update                                  uk.gov.hmrc.eusubsidycompliance.controllers.UndertakingController.updateSubsidy()
+POST       /subsidy/remove                                  uk.gov.hmrc.eusubsidycompliance.controllers.UndertakingController.updateSubsidy()
```
Comment (priyankadhiman3006) [inline conf/app.routes:12]: Shouldn't we call method  as removeSubsidy in here  like uk.gov.hmrc.eusubsidycompliance.controllers.UndertakingController.removeSubsidy()?

File: app/uk/gov/hmrc/eusubsidycompliance/models/NonHmrcSubsidy.scala [hunk 2]
```diff
@@ -22,14 +22,14 @@ import play.api.libs.json.{Json, OFormat}
...
 case class NonHmrcSubsidy(
-  subsidyUsageTransactionId: Option[SubsidyRef], // TODO if we can't get SCP09 sorted we'll need to uppercase ID in the format
-  allocationDate: LocalDate,
-  submissionDate: LocalDate,
-  publicAuthority: Option[String], // this shouldn't be optional, is required in create API but not retrieve
-  traderReference: Option[TraderRef],
-  nonHMRCSubsidyAmtEUR: SubsidyAmount,
-  businessEntityIdentifier: Option[EORI],
-  amendmentType: Option[EisSubsidyAmendmentType] = Option.empty // this only used for create
+                           subsidyUsageTransactionID: Option[SubsidyRef], // TODO if we can't get SCP09 sorted we'll need to uppercase ID in the format
```
Comment (priyankadhiman3006) [inline app/uk/gov/hmrc/eusubsidycompliance/models/NonHmrcSubsidy.scala:25]: I think the formatting has gone wrong here

File: conf/app.routes (hunk 1 above)
Comment (sam-miele) [reply conf/app.routes]: Nahh the update operation is the same. Just depends on the amendmenttype passed up
1 - add
2 - update
3 - remove

I've removed it to get rid of the ambiguity.

**Review by priyankadhiman3006 — APPROVED** [review]

**Review by carwynellis — COMMENTED** [review]
LGTM 👍

**Pull request merged**